includes/class-wc-stripe-helper.php

```diff
@@ -443,6 +443,25 @@ public static function get_order_by_setup_intent_id( $intent_id ) {
 		return false;
 	}
 
+	/**
+	 * Sanizite and retrieve a shortened statement descriptor.
+	 *
+	 * The descriptor will be used as prefix and the order ID will be concatenated, acting as a suffix.
+	 *
+	 * @param string $statement_descriptor
+	 * @param WC_Order $order
+	 * @return string $statement_descriptor Final shortened statement descriptor
+	 */
+	public static function get_shortened_statement_descriptor( $statement_descriptor = '', $order = null ) {
+		$statement_descriptor = self::clean_statement_descriptor( $statement_descriptor );
+
+		if ( method_exists( $order, 'get_id' ) && ! empty( $order->get_id() ) ) {
+			$statement_descriptor = $statement_descriptor . '* #' . $order->get_id();
```

Comment:
I think we'll want `$order->get_order_number()` – the ID is the post ID corresponding to the stored order but order number _can_ be different (being filterable) and is what's presented to the customer (e.g. see [template](https://github.com/woocommerce/woocommerce/blob/747cb6b7184ba9fdc875ab104da5839cfda8b4be/plugins/woocommerce/templates/myaccount/orders.php#L49)).

Reply:
Good catch, fixed in eaa6b2b.